app/controllers/offer/search_label.js

```diff
@@ -0,0 +1,74 @@
+import searchLabelController from "../search_label";
+
+export default searchLabelController.extend({
+  // ----- Arguments -----
+  packageType: null,
+  item: null,
+
+  //--Ajax Methods
+  createItem() {
+    const item = this.get("store").createRecord("item", this.itemParams());
+    this.runTask(
+      item.save().then(itemResponse => {
+        this.set("item", itemResponse);
+        this.addPackage();
+      })
+    );
+  },
+
+  addPackage() {
+    let pkgRecord = this.store.createRecord("package", this.packageParams());
+    this.runTask(
+      pkgRecord.save().then(pkg => {
+        this.transitionToRoute("receive_package", pkg.id, {
+          queryParams: { isUnplannedPackage: true }
+        });
+      })
+    );
+  },
+
+  donorConditions: Ember.computed(function() {
+    return this.get("store").peekAll("donorCondition");
+  }),
+
+  //--Params helpers
+  itemParams() {
+    const conditions = this.get("donorConditions");
+    const defaultDonorCondition =
+      conditions.filterBy("name", "Lightly Used").get("firstObject") ||
+      conditions.get("firstObject");
+    return {
+      offer: this.get("model"),
+      donorCondition: defaultDonorCondition,
+      state: "accepted",
+      packageType: this.get("packageType")
+    };
+  },
+
+  packageParams() {
+    const item = this.get("item");
+    const pkgType = this.get("packageType");
+    const packageTypeId = pkgType.id;
+
+    return {
+      notes: item.get("packageType.name"),
+      quantity: 1,
+      packageTypeId,
+      packageType: pkgType,
+      offerId: item.get("offer.id"),
+      item: item
+    };
+  },
+
+  // ----- Actions -----
+  actions: {
+    cancelSearch() {
+      window.history.back();
+    },
+
+    assignItemLabel(type) {
```

Comment:
Can we change the action name?
It's creating an item

Reply:
You are right, I am extending this controller from `search_label` controller to avoid code duplication and also shared its `hbs` file. So, there it has `assignItemLabel` we has name according to its task i.e assigning type to Item. Here in this controller I am overriding this action of creating Item and assigning. 
So, yeah had to do this just to avoid large code duplication.

Reply:
You can just create another action.